integration/setup_test.go

```diff
@@ -31,22 +31,24 @@ func (s *integrationTestSuite) Setup() {
 		o := obj
 		// check if object already exists
 		var existingObj client.Object
-		_ = integration.Client.Get(ctx, client.ObjectKey{
+		err := integration.Client.Get(ctx, client.ObjectKey{
 			Namespace: o.GetNamespace(),
 			Name:      o.GetName(),
 		}, existingObj)
 
-		if existingObj == nil {
+		if errors.IsNotFound(err) {
 			// if not create one
-			err := integration.Client.Create(ctx, &o)
+			err = integration.Client.Create(ctx, &o)
 			s.Require().NoError(err)
 
-			s.T().Log("created: ", o.GroupVersionKind().String(),
+			s.T().Log("Created: ", o.GroupVersionKind().String(),
 				o.GetNamespace()+"/"+o.GetName())
 
 			if o.GetKind() == "Deployment" {
 				deployments = append(deployments, o)
 			}
+		} else {
+			s.T().Log("Found: ", fmt.Sprintf("object: %v", o)+fmt.Sprintf("existing object: %v", existingObj))
```

Comment:
```suggestion
            s.Require().NoError(err)
			s.T().Log("Already exists: ", o.GroupVersionKind().String(),
				o.GetNamespace()+"/"+o.GetName())
```

`s.Require().NoError(err)`: because we want to ensure that it's err == nil unless `errors.IsNotFound(err)`.

`s.T().Log("Already exists: ", o.GroupVersionKind().String(),
				o.GetNamespace()+"/"+o.GetName())` : Because I think printing the entire object `o` object would make the logs very clutter-y. GVK + Namespace + Name should be optimally enough amount of info. around any resource to be stdout'd. What do you think :)